### DISCUSSION TITLE
Why the choice of toml

### DISCUSSION THREAD

**Discussion by noahfraiture** [helix-editor/helix#10831] [Q&A]
Hello,
I pretty much discovered the toml with helix and I don't understand the choice. At the beginning it's pretty great, it's pretty and nice but it quickly become ugly and hard to read. I would vote in favor of yaml. It's ok in most of the configuration file, but the languages file quickly become pretty big if you use different languages on a daily basis

I would argue that even the creator of toml said it's a bad thing and doesn't respect the DRY principle https://hitchdev.com/strictyaml/why-not/toml/

So was wondering, is there a deeper reason to have choose the toml configuration?

**Comment by kirawi** [comment]
Configuration with TOML was always intended to be a stopgap solution until the plugin system landed. As for why TOML, I think the only person who could answer that at this point is archseer because Helix shipped with TOML configuration from the first public release.

See also https://github.com/helix-editor/helix/discussions/10441

**Reply by pascalkuthe** [reply]
Yaml is also a horrible language. For simple configuration toml is great and widespread I the rust ecosystem (even the official tool chain uses it for configuration)

**Comment by a3lem** [comment]
I have nightmares about all the weird ways YAML parses ostensibly simple values. TOML took me a while to grok – longer than YAML, I'll admit. But now I use it everywhere. One of the reasons I like it is that, since it doesn't exactly lend itself to complex configs with multiple levels of nesting, it subtly steers developers (or me, at least) toward keeping config simpler.

**Comment by xtrasmal** [comment]
> yaml sounds like the word "jammel", 
> which almost sounds like "jammer"...
> which translate to "so sad, too bad" 


Jokes aside... if Helix's default config wasn't that good, I would have wanted to split my configuration into multiple files.  
At this point I have 60 lines of config, which I hardly every touch. So that is manageable. Apart from the config, I have about the same amount of line in languages.toml. The defaults are a great deal. 

I do use other tools, that use toml as well. I like the format when there isn't much to configure. So at this point there is no problem, so the url is out of context. if you would manage many docker/ansible configs then yaml makes sense.

Btw..The DRY principle.. such a under explained and understood principle. @noahfraiture You should read this: **[DRY is about knowledge](https://verraes.net/2014/08/dry-is-about-knowledge/)**
